### PR TITLE
Support to refresh existing protection source

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
1) Updated module to support source refresh.
2) New parameter 'refresh' is added based on which sources will be refreshed.

### CODE REVIEW

**Review by pyashish — COMMENTED** [review]
Looks good, do have clear documentation about having refresh vs update vs create ?

File: docs/modules/cohesity_migrate_vm.rst [hunk 1]
```diff
@@ -62,6 +62,14 @@ Parameters
     If this is set to true, then the network will be detached from the recovered VMs. All the other networking parameters set will be ignored if set to true
 
 
+  cluster_compute_resource (optional, str, None)
```
[the remark this answers is not in the document]
Comment (naveena-maplelabs) [reply docs/modules/cohesity_migrate_vm.rst:65]: This documentation is for previous enhancement, was missed earlier.

File: playbooks/physical-workflow/refresh_physical_sources.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        username: "{{ cohesity_username }}"
+        password: "{{ cohesity_password }}"
+        validate_certs: "{{ cohesity_validate_certs }}"
+        state: present
+        endpoint: "{{ item }}"
+        environment: "Physical"
+        refresh: true
+      with_items: "{{ linux_servers }}"
+
+      # If the list of sources to be registered is available in the inventory
```
[the remark this answers is not in the document]
Comment (naveena-maplelabs) [reply playbooks/physical-workflow/refresh_physical_sources.yml:24]: okay

File: playbooks/vmware-workflow/refresh_source.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Playbook to refresh Vmware source in the cluster.
+---
+- hosts: localhost
+  become: false
+  collections:
+    - cohesity.dataprotect
+  tasks:
+    - name: "Cohesity Protection Source: Refresh VMware resource"
```
[the remark this answers is not in the document]
Comment (naveena-maplelabs) [reply playbooks/vmware-workflow/refresh_source.yml:8]: refresh field in playbook

**Pull request merged**